### PR TITLE
Add some checks on login

### PR DESCRIPTION
**Pull request by Blackmane**

### CODE REVIEW

File: src/app/Application.cpp [hunk 1]
```diff
@@ -72,6 +75,12 @@ void Application::handleLoginSuccess(const QString &token,
     m_req.requestGuilds();
 }
 
+void Application::handleLoginFailed(const QString &errorNumber, const QString &errorMessage)
+{
+    QString title = "Error(" + errorNumber + ")";
```
Comment (guerinoni) [inline src/app/Application.cpp:80]: Prefer `const auto` when is possible

File: src/handlers/Authenticator.cpp [hunk 2]
```diff
@@ -4,8 +4,10 @@
 
 #include <QJsonDocument>
 #include <QJsonObject>
+#include <QJsonArray>
 #include <QNetworkReply>
 #include <QSettings>
+#include <QString>
```
Comment (guerinoni) [inline src/handlers/Authenticator.cpp:10]: This can be removed

File: src/handlers/Authenticator.cpp [hunk 3]
```diff
@@ -19,16 +21,49 @@ void Authenticator::requestLogin(QString email, QString pass, QString twoFA)
     QJsonDocument doc(data);
 
     const auto reply = m_nam.post(req, doc.toJson());
-    connect(reply, &QNetworkReply::finished, this, [&]() {
+    connect(reply, &QNetworkReply::finished, this, [&, twoFA]() {
```
Comment (guerinoni) [inline src/handlers/Authenticator.cpp:24]: Change `[&, twoFA]` with `[this, twoFA]` because `&` capture all by reference

File: src/qml/qml.qrc [hunk 4]
```diff
@@ -9,5 +9,6 @@
         <file>UserInfoBar.qml</file>
         <file>main.qml</file>
         <file>qtquickcontrols2.conf</file>
+        <file>PopupMessage.qml</file>
```
Comment (guerinoni) [inline src/qml/qml.qrc:12]: Order alphabetically (right click on qtCreator and "sort alphabetically")

File: src/qml/main.qml [hunk 5]
```diff
@@ -51,5 +51,12 @@ ApplicationWindow {
         function onLoginSuccess() {
             stackview.push(chat)
         }
+        function onShowError(title, errorMessage) {
+            var popupComponent = Qt.createComponent("qrc:/PopupMessage.qml")
+            var popupMessageError = popupComponent.createObject(stackview);
+            popupMessageError.open()
+            popupMessageError.title = qsTr(title)
```
Comment (guerinoni) [inline src/qml/main.qml:58]: I don't know if this can work for translations...

**Review by guerinoni — CHANGES_REQUESTED** [review]

File: src/qml/PopupMessage.qml [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import QtQuick 2.8;
+import QtQuick.Controls 2.1;
```
Comment (guerinoni) [inline src/qml/PopupMessage.qml:2]: Keep this version up to date

File: src/qml/main.qml (hunk 5 above)
Comment (Blackmane) [reply src/qml/main.qml:58]: How can i improve it?

Comment (guerinoni) [reply src/qml/main.qml:58]: Try to translate and see if it works... For now we can also stay only with english

File: src/qml/PopupMessage.qml [hunk 7]
```diff
@@ -0,0 +1,36 @@
+import QtQuick 2.15;
+import QtQuick.Controls 2.15;
+import QtQuick.Controls.Material 2.15
+
+Popup {
+    property alias message : error_message.text
+    property alias title : label.text
+
+    id: popup;
```
Comment (guerinoni) [inline src/qml/PopupMessage.qml:9]: `id` property should be first property and separated with a blank line

**Review by guerinoni — CHANGES_REQUESTED** [review]

File: src/qml/main.qml (hunk 5 above)
Comment (Blackmane) [reply src/qml/main.qml:58]: Oh I think I understand the problem. I used shortsightedly qsTr.

File: src/qml/PopupMessage.qml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    Text {
+        anchors {
+            top: login.bottom
+            topMargin: 10
+            horizontalCenter: popup.horizontalCenter
+        }
+
+        id: error_message
```
Comment (guerinoni) [inline src/qml/PopupMessage.qml:31]: Also here

File: src/qml/PopupMessage.qml [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+    anchors.centerIn: parent
+    width: 400;
+    height: 200;
+    modal: true;
+    focus: true;
+    closePolicy: Popup.CloseOnEscape | Popup.CloseOnPressOutside
+
+    Label {
+        id: label
```
Comment (guerinoni) [inline src/qml/PopupMessage.qml:19]: And here

**Review by guerinoni — CHANGES_REQUESTED** [review]

**Comment by guerinoni** [comment]
Many thanks!

**Pull request merged**